packages/terra-functional-testing/src/reporters/fileOutput-reporter/index.js

```diff
@@ -0,0 +1,220 @@
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  setTestModule(specsValue) {
+    const index = specsValue.lastIndexOf(`packages${path.sep}`);
+    if (index > -1) {
+      const testFilePath = specsValue.substring(index).split(path.sep);
+      const moduleName = testFilePath && testFilePath[1]
+        ? testFilePath[1]
+        : process.cwd().split(path.sep).pop();
+      if (moduleName && moduleName !== this.moduleName) {
+        this.moduleName = moduleName;
+      }
+    }
+  }
+
+  onRunnerEnd(runner) {
+    this.runners.push(runner);
+    this.printReport(runner);
+  }
+
+  getMessage(runner) {
+    const preface = `[${this.getEnviromentCombo(
+      runner.capabilities,
+      false,
+      runner.isMultiremote,
+    ).trim()} #${runner.cid}]`;
+    const divider = '------------------------------------------------------------------';
+    const results = this.getResultDisplay();
+    if (results.length === 0) {
+      return null;
+    }
+    const testLinks = runner.isMultiremote
+      ? Object.entries(runner.capabilities).map(
+        ([instanceName, capabilities]) => this.getTestLink({
+          config: { ...runner.config, ...{ capabilities } },
+          sessionId: capabilities.sessionId,
+          isMultiremote: runner.isMultiremote,
+          instanceName,
+        }),
+      )
+      : this.getTestLink(runner);
+    const output = [
+      ...this.getHeaderDisplay(runner),
+      '',
+      ...results,
+      ...this.getFailureDisplay(),
+      ...(testLinks.length ? ['', ...testLinks] : []),
+    ];
+    const prefacedOutput = output.map((value) => (value ? `${preface} ${value}` : preface));
+    return `${divider}\n${prefacedOutput}`;
+  }
+
+  printReport() {
+    const { runners } = this;
+    if (runners && runners.length) {
+      runners.forEach((runner, index) => {
+        // determine correct file name given configuration for run
+        if (index === 0) {
+          const { cid, capabilities, config } = runner;
+          const { browserName } = capabilities;
+          this.fileNameCheck(config.launcherOptions, browserName, cid);
+        }
+
+        this.setTestModule(runner.specs[0]);
+
+        if (!this.resultJsonObject.output[this.moduleName]) {
+          this.resultJsonObject.output[this.moduleName] = [];
+        }
+        let readableMessage = this.getMessage(runner);
+        const readableArrayMessage = readableMessage.toString().split(',');
+        for (let i = 0; i < readableArrayMessage.length; i + 1) {
+          readableArrayMessage[i] = stripAnsi(`${readableArrayMessage[i]}${endOfLine}`);
+        }
+        readableMessage = readableArrayMessage.join('');
```

Comment:
how about being declarative and using map:
```suggestion
        const messages = readableArrayMessage.map((message) => stripAnsi(`${message}${endOfLine}`).join('');
```

Reply:
https://github.com/cerner/terra-toolkit/pull/662/files#diff-5aca2141ee738df380d2de0da498a0a17727d935d42de74bbbb00a3046b1febeR168-R169